### ISSUE TITLE
Author/editor responsibility for Zenodo upload + codecheck.yml update

### ISSUE THREAD

**Issue by mstimberg** [codecheckers/discussion#21]
This is possibly biased by my [ReScience C](https://rescience.github.io/) experience, but wouldn't it make sense to have the final steps that formalize the CODECHECK, i.e. the upload to Zenodo and the update of the `codecheck.yml` to be rather an editor and not a reviewer task? If I am not mistaken, so far @sje30 and @nuest have been both editors and reviewers at the same time, so this question did not really arise yet. In my opinion, having these final "validation" tasks been done by editors could avoid incorrect (in the sense of missing files, metadata, etc.) code checks being uploaded to Zenodo and in general ensure consistency. I could also imagine that in the future you might decide to e.g. have additional metadata in the Zenodo deposit – it would be easier if you could add them yourself instead of asking reviewers to do it. Of course, ideally you'd have a bot like JOSS's [whedon](https://joss.readthedocs.io/en/latest/whedon.html) do these tasks for you :)

**Comment by nuest** [comment]
Yes, a CODECHECK-whedon would be awesome, and we started collecting some ideas, too: #6 #13 #20 

You're spot on with your analysis that so far we always were checker+editor in one person. I think you make a good point that a second pair of eyes makes sense here. However, it might also require the editor to render the certificate document, or the editor must tell the codechecker the to be expected DOI/citation of the report. I'd prefer the latter, because we give the codecheckers freedom to choose whatever tool they want for their report.

@sje30 Thoughts? Should we let the CODECHECK editor reserve the DOI and create the Zenodo record?

**Comment by mstimberg** [comment]
Good points, sorry for coming into this with a bit of naïve views :) Maybe one approach could be that when a codechecker agrees to do the review, the editor clones the repository, reserves a DOI and creates/updates a first `codecheck.yml` file with the registry number and the DOI? This way, all the formal stuff would be already fixed and the code checker could take it from there. At the end of the process (with a `codecheck.pdf` generated by the codechecker in the repository), the editor would then upload everything to Zenodo.

**Comment by nuest** [comment]
Sounds good to me. I'll put it in my "queue" to work this into the community workflow guide and will ping here when it is done.

@mstimberg For your current review (https://github.com/codecheckers/register/issues/35), I think we have to stick to the current plan though and you go ahead with the publication of the Zenodo record.

**Comment by mstimberg** [comment]
> @mstimberg For your current review ([codecheckers/register#35](https://github.com/codecheckers/register/issues/35)), I think we have to stick to the current plan though and you go ahead with the publication of the Zenodo record.

Sure, no problem.

**Comment by nuest** [comment]
Revisiting this after a long while, with the following points:

- Revising the Zenodo metadata becomes easier with the new features for communities - we can now edit metadata of records submitted to our community.
- Situation for OSF remains: editor needs access to the project.

With the limited resources we have, we currently won't add tasks to the editor.
